src/containers/CreateArticle/CreateArticlePage.js

```diff
@@ -0,0 +1,254 @@
+      description: "",
+      error: {
+        title: "",
+        body: ""
+      },
+      message: ""
+    };
+  }
+
+  clearErrorMessages = () => {
+    this.setState({
+      error: {
+        title: "",
+        body: ""
+      }
+    });
+  };
+
+  titleChangeHandler = title => {
+    this.clearErrorMessages();
+    this.setState({
+      title
+    });
+  };
+
+  editorChangeHandler = body => {
+    this.clearErrorMessages();
+    this.setState({
+      body
+    });
+  };
+
+  titlePlaceholderFocusInHandler = () => {
+    const { title } = this.state;
+    if (title.trim() === "Title") {
+      this.setState({ title: "" });
+    }
+  };
+
+  titlePlaceholderFocusOutHandler = () => {
+    const { title } = this.state;
+    if (title.trim() === "") {
+      this.setState({
+        title: "Title"
+      });
+    }
+  };
+
+  bodyPlaceholderFocusInHandler = () => {
+    const { body } = this.state;
+    if (
+      body.trim() === "Share your thoughts..." ||
+      body.trim() === "<p>Share your thoughts...</p>"
+    ) {
+      this.setState({
+        body: ""
+      });
+    }
+  };
+
+  bodyPlaceholderFocusOutHandler = () => {
+    const { body } = this.state;
+    if (body.trim() === "") {
+      this.setState({
+        body: "Share your thoughts..."
+      });
+    }
+  };
+
+  imageUploadHandler = (blobInfo, success, failure) => {
+    try {
+      const formData = new FormData();
+      formData.append("file", blobInfo.blob());
+      formData.append("upload_preset", "asgjqcgx");
+      formData.append("api_key", "329369412183662");
+      const uploadUrl =
+        "https://api.cloudinary.com/v1_1/dn6fnuhxr/image/upload";
+      return axios
+        .post(uploadUrl, formData, {
+          headers: {
+            "Content-Type": "Application/x-www-form-urlencoded",
+            "X-Requested-With": "XMLHttpRequest"
+          }
+        })
+        .then(response => {
+          const data = response.data;
+          const fileUrl = data.secure_url;
+          success(fileUrl);
+        });
+    } catch (error) {
+      failure("file upload failed");
+    }
+  };
+
+  validateArticle = () => {
+    const { title, body } = this.state;
+    const bodyValue = this.rootNode.props.value;
+
+    if (title === "Title" || title.trim() === "") {
+      this.setState({
+        error: {
+          title: "Publishing will be available when you provide a title"
+        }
+      });
+      return false;
+    }
+
+    if (
+      body === "Share your thoughts..." ||
+      bodyValue.length < 30 ||
+      body.trim() === ""
+    ) {
+      this.setState({
+        error: {
+          body: "Publishing will be available when you start typing"
+        }
+      });
+      return false;
+    }
+
+    this.setState({
+      error: {}
+    });
+
+    return true;
+  };
+
+  handleSubmit = () => {
+    const { title, body, error } = this.state;
+    const { createArticle } = this.props;
+    const description = stripHtml(body).slice(0, 200);
```

Comment:
Nice work on generating this description from the body. On line 151, I think having 200 words (if I am correct on what I'm seeing) is a bit too much. In my opinion, the 30 first characters (or 25) from the body should be enough for the description.

Reply:
It gets the first 200 characters ie letters, spaces, etc

Reply:
Oh! my bad. 200 characters is cool. I thought you were generating words, not characters.